### PR TITLE
fix(jans-cedarling): Fix cedarling to handle iss in different cases

### LINKED ISSUE #12943: fix(jans-cedarling): Fix cedarling to handle domains not only in lower case
**Describe the bug**

When we use TrustedIssuer and url in `openid_configuration_endpoint` field. It is parsed by `url` lib. And in result we have lowercase URL . And we use the index over lowercase URL origin.

But if real url isn't lowercased and in JWT issuer is not lowercased, we can't find the validator.

**To Reproduce**
I found the bug when trying to understand the problem when debugging `"openid_configuration_endpoint": ""https://KBTG-GluuFlex.azlabs.sg/.well-known/openid-configuration"`

**Expected behavior**
JWT with `  "iss": "https://KBTG-GluuFlex.azlabs.sg"` should be handled correctly.

**Possible solution**
Before searching validators parse the issuer from JWT using `url` lib.

**Additional context**
The problem was found in `v1.14.0`

### PR DESCRIPTION
**Pull request by olehbozhok**
…es (upper and lower)

### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

#### Target issue
  
  
  
  
closes #12943

#### Implementation Details
  
   Trusted issuer URL from config is parsed by `url` module.  So I added parse `iss` using `url` crate to be consistent.
-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced issuer normalization to properly lowercase hosts in URLs while maintaining backward compatibility for non-URL strings.

* **Tests**
  * Expanded test coverage for issuer normalization edge cases, including uppercase host handling and URL parsing scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by mo-auto** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/janssenproject/pr-checks/ccf86115-ce69-41af-b19a-4f83447deb73?refs=849405573df52f1e37f6b5211613e41f39790966&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: jans-cedarling/cedarling/src/common/issuer_utils.rs [hunk 1]
```diff
@@ -7,15 +7,27 @@
...
+    use test_utils::assert_eq;
 
     #[test]
     fn test_normalize_issuer() {
+        // Test that uppercase letters are lowered
+        assert_eq!(
+            normalize_issuer("https://DEV-1e737fn2gji0j3fe.us.auth0.com/"),
+            "https://dev-1e737fn2gji0j3fe.us.auth0.com"
+        );
+
```
[the remark this answers is not in the document]
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/issuer_utils.rs:30]: Yes, it was intentional. It is OK to use `use test_utils::assert_eq;`

**Comment by olehbozhok** [comment]
@coderabbitai can you approve?
